test/test.js

```diff
@@ -106,6 +106,162 @@ describe('timeout()', function () {
     })
   })
 
+  describe('req.resetTimeout()', function () {
+    it('should reset the timeout', function (done) {
+      var server = createServer(null,
+        function (req, res) { req.resetTimeout(1000) },
+        function (req, res) {
+          assert.ok(!req.timedout)
+          res.end('Hello')
+        })
+
+      request(server)
+      .get('/')
+      .expect(200, 'Hello', done)
+    })
+    it('should reset the timeout with a string', function (done) {
+      var server = createServer(null,
+        function (req, res) { req.resetTimeout('1s') },
+        function (req, res) {
+          assert.ok(!req.timedout)
+          res.end('Hello')
+        })
+
+      request(server)
+      .get('/')
+      .expect(200, 'Hello', done)
+    })
+    it('should reset the timeout with 5000 default', function (done) {
+      var server = createServer(null,
+        function (req, res) { req.resetTimeout() },
```

Comment:
How does this test validate that 5000 is what the reset used? It is identical to the first test just missing the argument.

It seems like if the impl was change to not accept an argument, all three of these tests still pass?

Reply:
Good call. I updated the tests to test the actual timeout.